tools/makerootfs.sh

```diff
@@ -1,30 +1,118 @@
 #!/bin/bash
 # Usage:
 #
-#     ./makerootfs.sh <image.yml> <output rootfs image> <fs> <arch>
+#     ./makerootfs.sh mode [-y <image.yml>] [-i <output rootfs image>] [-t <tar file>] [-f <filesystem format>] [-a <arch>]
 # <fs> defaults to squash
 # <arch> defaults to the current machine architecture
 
 set -e
 set -o pipefail
 
+# mode 3 - generate image from yml
+do_image() {
+  # shellcheck disable=SC2166
+  if [ -z "$ymlfile" -o -z "$imgfile" -o -n "$tarfile" ]; then
+    echo "must supply ymlfile and imgfile and not tarfile" >&2
+    help
+  fi
+  # did we specify an architecture?
+  ARCHARG=""
+  if [ -n "$arch" ]; then
+    ARCHARG="-arch ${arch}"
+  fi
+  : > "$IMAGE"
+  # shellcheck disable=SC2086
+  linuxkit build -docker ${ARCHARG} -o - "$ymlfile" | docker run -i --rm -v /dev:/dev --privileged -v "$IMAGE:/rootfs.img" "${MKROOTFS_TAG}"
+}
+
+# mode 1 - generate tarfile from yml and save
+do_tar() {
+  if [ -z "$ymlfile" -o -z "$tarfile" -o -n "$imgfile" ]; then
```

Comment:
FWIW I've  been following the shellcheck suggestion for these, just to make it shut up. That is easier than adding the shellcheck disable comments.

Reply:
I am a bit of a stickler here, as the syntax I used is 100% valid POSIX sh, whereas the suggested one is only valid in bash. I get that we are using bash, but I don't think we should move from more universal to less universal without a good reason.